packages/@azure/servicebus/data-plane/test/renewLock.spec.ts

```diff
@@ -10,176 +10,677 @@ chai.use(chaiAsPromised);
+    );
+  }
+  if (
+    !process.env.TOPIC_NAME ||
+    !process.env.TOPIC_NAME_NO_PARTITION ||
+    !process.env.TOPIC_NAME_NO_PARTITION_SESSION ||
+    !process.env.TOPIC_NAME_SESSION
+  ) {
+    throw new Error(
+      "Define TOPIC_NAME, TOPIC_NAME_NO_PARTITION, TOPIC_NAME_SESSION & TOPIC_NAME_NO_PARTITION_SESSION in your environment before running integration tests."
+    );
+  }
+  if (
+    !process.env.QUEUE_NAME ||
+    !process.env.QUEUE_NAME_NO_PARTITION ||
+    !process.env.QUEUE_NAME_NO_PARTITION_SESSION ||
+    !process.env.QUEUE_NAME_SESSION
+  ) {
+    throw new Error(
+      "Define QUEUE_NAME, QUEUE_NAME_NO_PARTITION, QUEUE_NAME_SESSION & QUEUE_NAME_NO_PARTITION_SESSION in your environment before running integration tests."
+    );
+  }
+  if (
+    !process.env.SUBSCRIPTION_NAME ||
+    !process.env.SUBSCRIPTION_NAME_NO_PARTITION ||
+    !process.env.SUBSCRIPTION_NAME_NO_PARTITION_SESSION ||
+    !process.env.SUBSCRIPTION_NAME_SESSION
+  ) {
+    throw new Error(
+      "Define SUBSCRIPTION_NAME, SUBSCRIPTION_NAME_NO_PARTITION, SUBSCRIPTION_NAME_SESSION & SUBSCRIPTION_NAME_NO_PARTITION_SESSION in your environment before running integration tests."
+    );
+  }
+
+  namespace = Namespace.createFromConnectionString(process.env.SERVICEBUS_CONNECTION_STRING);
+
+  testMessage = {
+    body: `hello-world-1 : ${generateUuid()}`,
+    messageId: generateUuid()
+  };
+
+  // Partitioned Queues and Subscriptions
+  partitionedQueueClient = namespace.createQueueClient(process.env.QUEUE_NAME);
+
+  // Unpartitioned Queues and Subscriptions
+  unpartitionedQueueClient = namespace.createQueueClient(process.env.QUEUE_NAME_NO_PARTITION);
+}
 
-    if (!process.env.SERVICEBUS_CONNECTION_STRING) {
-      throw new Error(
-        "Define SERVICEBUS_CONNECTION_STRING in your environment before running integration tests."
+async function afterEachTest(): Promise<void> {
+  await namespace.close();
+}
+
+let uncaughtErrorFromHandlers: Error | undefined;
+
+const onError: OnError = (err: MessagingError | Error) => {
+  uncaughtErrorFromHandlers = err;
+};
+
+function assertTimestampsAreApproximatelyEqual(
+  actualTimeInUTC: Date | undefined,
+  expectedTimeInUTC: Date,
+  label: string
+): void {
+  if (actualTimeInUTC) {
+    should.equal(
+      Math.pow((actualTimeInUTC.valueOf() - expectedTimeInUTC.valueOf()) / 1000, 2) < 1,
+      true,
+      `${label}: Actual time ${actualTimeInUTC} must be approximately equal to ${expectedTimeInUTC}`
+    );
+  }
+}
+
+//
+
+// Tests for Lock Renewal, see -  https://github.com/Azure/azure-service-bus-node/issues/103
+// Receive a msg using Batch Receiver, test renewLock()
+async function testBatchReceiverManualLockRenewalHappyCase(
+  senderClient: QueueClient | TopicClient,
+  receiverClient: QueueClient | SubscriptionClient
+): Promise<void> {
+  await senderClient.send(testMessage);
+
+  await delay(5000);
```

Comment:
This delay between sending and receiving is not needed, since the lock duration only starts once it is received. Right?

Reply:
That's right, I think I added it while debugging topic subscriptions issue. Removing